### DISCUSSION TITLE
Simple Open / Toggle Closed By HMR?

### DISCUSSION THREAD

**Discussion by tiagoapp** [sveltejs/kit#6132] [Q&A]
Say you click on the button to open the message... edit some code inside the panel and the open state will revert to false and close the panel.

Just used to React and Vue or the state is kept alive and makes it easy to update conditional blocks.
Is this normal / expected behavior on Svelte Kit or am I missing something?

(tested with very last and fresh sveltekit from today)

```svelte
<script lang="ts">
let open = false
</script>

{#if open}
<div class="panel">
    <h1>hello</h1>
</div>
{/if}

<button on:click={() => open = true}>open</button>
```

**Comment by dominikg** [comment]
https://kit.svelte.dev/faq#hmr
https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/faq.md#why-is-component-state-reset-on-hmr-update
https://github.com/sveltejs/svelte-hmr/blob/master/packages/svelte-hmr#preservation-of-local-state
